File: adventofcode/2021/day1/day12.cpp
#include <bits/stdc++.h>
using namespace std;
/*
part 2
*/
int main() {
    freopen("inputs/input1.txt", "r", stdin);
    freopen("outputs/output2.txt", "w", stdout);
    string tmp;
    int cnt = 0, depth;
    vector<int> depths;
    while(getline(cin, tmp)) {
        depth = stoi(tmp); 
        depths.push_back(depth);
    }
    for(int i=3; i<depths.size(); i++) {
        cnt += (depths[i]>depths[i-3]);
    }
    cout<<cnt<<endl;
}

File: adventofcode/2021/day1/day12.py
"""
Part 2
"""
if __name__ == "__main__":
    with open("inputs/input1.txt", "r") as f:
        data = list(map(int, f.read().splitlines()))
        print(sum(1 for num1, num4 in zip(data, data[3:]) if num4>num1))
